src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -0,0 +1,183 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codeu.controller;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+
+import codeu.model.data.User.Sex;
+import codeu.model.data.User;
+import codeu.model.data.User.Occupation;
+import codeu.model.store.basic.UserStore;
+
+public class ProfileServlet extends HttpServlet {
+	private UserStore userStore;
+	
+	BlobstoreService blobstoreService;
+	ImagesService imagesService;
+	
+	@Override
+	public void init() throws ServletException {
+		super.init();
+		userStore = UserStore.getInstance();
+		blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+		imagesService = ImagesServiceFactory.getImagesService();
+	}
+
+	@Override
+	public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException{
+		String profileEditName = getNameFromURL(request.getRequestURL());
+		
+		User user = userStore.getUser(profileEditName);
+		UserStore userStore = UserStore.getInstance();
+
+		try {
+			Date newBirthday = parseDate(request.getParameter("updated birthday"));
+			user.setBirthday(newBirthday);
+		} catch(NullPointerException e) {
+			System.out.println("Didn't set birthday");
+		}
+		
+		try {
+			String newDescription = request.getParameter("updated description");
+			
+			if(newDescription.trim() != "") {
+				user.setDescription(newDescription);
+			}
+		} catch(NullPointerException e) {
+			System.out.println("Didn't set description");
+		}
+		
+		try {
+			String newSex = request.getParameter("updated sex");
+			user.setSex(Sex.valueOf(newSex));
+		} catch(NullPointerException e) {
+			System.out.println("Didn't set sex");
+		}
```

Comment:
For this sequence of try and catch statements I think it may be better to explicitly check if the field is non-null instead of catching a null pointer exception. For example:
```java
if (request.getParameter('updated birthday') != null) {
  user.setBirthday(parseDate(request.getParameter('updated birthday'));
}